### DISCUSSION TITLE
Issue when installing madCAD

### DISCUSSION THREAD

**Discussion by dalopilo** [jimy-byerley/pymadcad#66] [General]
Hello everyone, 

I tried to install madcad on my Ubuntu for ARM server Virtual Machine, but I am not sure that it worked. 
I tried to launch the bearing.py script from the examples, and here is the issue I am facing :

<img width="1255" alt="Capture d’écran 2023-01-23 à 10 14 08" src="https://user-images.githubusercontent.com/79650467/214005134-f6c8259a-1ff8-42ed-a124-4004343307e1.png">

Do you know why nothing is appearing on the screen ?

Then, I tried to install the UIMadCAD package to get the GUI. I faced another issue :

<img width="710" alt="Capture d’écran 2023-01-23 à 10 15 55" src="https://user-images.githubusercontent.com/79650467/214005975-2761482e-da4f-4544-98ae-205b9525bea1.png">

Could you please help me, I need to be able to use this software for an important biomechanics project.

Thank you for your time and commitment !

**Comment by jimy-byerley** [comment]
Hello !

1. uimadcad

  the issue in installing uimadcad is nothing:  of course `dpkg` cannot finish the installation because some of the dependencies are not installed on the system yet. You just need to run `apt-get` afterward to finish the job and this should work
  ```bash
  sudo apt-get --fix-broken install
  ```

2. pymadcad

  For the first issue with the backtrace, the first error mentions `Requested OpenGL version 330, got 0`  meaning openGL failed to initialize. This is probably due to you running in a VM ... openGL needs a GPU driver or a CPU driver. 
  If you have a modern CPU with multiple cores, [mesa](https://mesa3d.org/)  can be that CPU driver, this will always be slower than with a GPU of course ...
  I did not tried it recently, but the following commandline may install it:
  ```bash
  sudo apt-get install mesa-utils
  ```

**Comment by dalopilo** [comment]
Hello !

Thank you for your quick reply. Instead of using the Ubuntu VM I tried installing everything directly on my M1 Pro MacBook Pro. I faced some issues. Could you please help me again ? 

￼


Thank you for your time and commitment, 

Irvin

**Reply by jimy-byerley** [reply]
What is this issue you are facing ?

I dont know it it is the same for your issue, but IOS has some ... let say *history* with OpenGL. Since Apple decided to feature their own proprietary GPU rendering pipeline (Metal), they decided to freeze support for OpenGL to version 2.x (last 2.x is from 2009, pymadcad requires version >=3.3).  As Metal was not widely adopted by all softwares (and could not be since being proprietary and not cross platform), they decided years after to upgrade support to openGL 4.x for programs written in Objective-C. We are still there today, and since pymadcad isn't written in Objective-C and Qt neither, it has poor chances to run on IOS ...

#33 is a good example of this issue with openGL.

Jimy

**Comment by wociety** [comment]
Hello.
I'm a newbie. Thanks for providing such a wonderful library. I would like to try pymadcad.
However, when trying to install pymadcad using pip in a Windows environment, the following error occurs.
-------------------------------------------------- --------------
  Preparing metadata (setup.py) ... error
   error: subprocess-exited-with-error

   × python setup.py egg_info did not run successfully.
   │ exit code: 1
   ╰─> [7 lines of output]
       Traceback (most recent call last):
         File "<string>", line 2, in <module>
         File "<pip-setuptools-caller>", line 34, in <module>
         File "C:\Users\wocie\AppData\Local\Temp\pip-install-4_jccb79\arrex_a51c1d5dec5b445daf2e2749254cc72c\setup.py", line 40, in <module>
           long_description = open('README.md').read(),
                              ^^^^^^^^^^^^^^^^^^^^^^^^
       UnicodeDecodeError: 'cp949' codec can't decode byte 0xe2 in position 3861: illegal multibyte sequence
       [end of output]

   note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
-------------------------------------------------- -------------------------

I guess this is an error reading the readme file. Is there any good way to solve this problem?

**Reply by jimy-byerley** [reply]
Weird issue, but that is a different topic than with IOS and openGL. Can I suggest we move to a new [issue](https://github.com/jimy-byerley/pymadcad/issues) ?

**Reply by jimy-byerley** [reply]
This seems to be an issue with the readme file encoding:  your OS might be set to use ascii encoding by default, and python is using that default.
This can easily be patched

**Reply by wociety** [reply]
Thank you for your quick answer. you can move to a new issue of course.

**Reply by wociety** [reply]
My Windows 11 is Korean language version and this might be the cause I guess.

**Reply by jimy-byerley** [reply]
I opened [arrex/#3](https://github.com/jimy-byerley/arrex/issues/3) for this topic
